js/notation.js

```diff
@@ -349,6 +349,7 @@ class Notation {
         } else {
             if (this.activity.logo.runningLilypond) {
                 obj = rationalToFraction(factor);
+                if (!obj) return; // Prevent undefined obj error
```

Comment:
Maybe print factor to console so we can diagnose why this is happening? Or maybe fix rationalToFraction?

Reply:
Ok I will Get on it right away and update accordingly

Reply:
Hello, even though i print factor in console, there seems to be some issue in the initial function as while doing so it throws errors, also i did find some potential problems in error handling in rationalToFraction function which are as follows-
1. Infinite loop risks are there in the while loop in edge cases
2. there are floating point inaccuracies which might cause very large denominators due to very small values of d
3. also there could be more improvements in error handling of edge cases such as limiting the maximum number of cases or the number possible

but if i do these changes in rationalToFraction function in js\utils\utils.js file can it cause issues in the software at other places as it is used in a lot of places, please guide me for the same @walterbender

Reply:
I think all of the changes you recommend for rationalToFraction make sense regardless of where it is used. Best to address the problem at its source.